pkg/dockerregistry/server/metrics/metrics.go

```diff
@@ -60,24 +81,45 @@ func (m *metricTimer) Stop() {
 	m.collector.WithLabelValues(m.labels...).Observe(float64(time.Since(m.startTime)) / float64(time.Second))
 }
 
-func newWrapper(reponame string) wrapped.Wrapper {
-	return func(ctx context.Context, funcname string, f func(ctx context.Context) error) error {
-		defer NewTimer(registryAPIRequests, []string{strings.ToLower(funcname), reponame}).Stop()
+// NewRepository wraps a distribution.Repository to collect statistics.
+func NewRepository(r distribution.Repository, reponame string) distribution.Repository {
+	return wrapped.NewRepository(r, func(ctx context.Context, funcname string, f func(ctx context.Context) error) error {
+		defer NewTimer(requestDurationSeconds, []string{strings.ToLower(funcname), reponame}).Stop()
 		return f(ctx)
-	}
+	})
 }
 
-// NewBlobStore wraps a distribution.BlobStore to collect statistics.
-func NewBlobStore(bs distribution.BlobStore, reponame string) distribution.BlobStore {
-	return wrapped.NewBlobStore(bs, newWrapper(reponame))
+// NewPullthroughRepository wraps distribution.Repository that is instantiated
+// for a remote registry to collect statistics.
+func NewPullthroughRepository(r distribution.Repository, registry string) distribution.Repository {
+	return wrapped.NewRepository(r, func(ctx context.Context, funcname string, f func(ctx context.Context) error) error {
+		labelValues := []string{strings.ToLower(registry), strings.ToLower(funcname)}
+		defer NewTimer(pullthroughRepositoryDurationSeconds, labelValues).Stop()
+		err := f(ctx)
+		if err != nil {
+			pullthroughRepositoryErrorsTotal.WithLabelValues(labelValues...).Inc()
+		}
+		return err
+	})
 }
 
-// NewManifestService wraps a distribution.ManifestService to collect statistics
-func NewManifestService(ms distribution.ManifestService, reponame string) distribution.ManifestService {
-	return wrapped.NewManifestService(ms, newWrapper(reponame))
+type RepositoryRetriever struct {
+	RepositoryRetriever registryclient.RepositoryRetriever
 }
 
-// NewTagService wraps a distribution.TagService to collect statistics
-func NewTagService(ts distribution.TagService, reponame string) distribution.TagService {
-	return wrapped.NewTagService(ts, newWrapper(reponame))
+func (rr RepositoryRetriever) Repository(ctx gocontext.Context, registry *url.URL, repoName string, insecure bool) (distribution.Repository, error) {
+	labelValues := []string{strings.ToLower(registry.Host), "init"}
```

Comment:
`registry.Host` is `registry-1.docker.io` for docker.io images. I'd like to see there the registry name from the docker reference, but this interface is from Origin.
/cc @legionus @miminar @bparees

Reply:
You could use the [(*DockerImageReference). DaemonMinimal()](https://github.com/openshift/origin/blob/master/pkg/image/apis/image/helper.go#L248) helper or do its equivalent.